pkg/apis/objectbucket.io/v1alpha1/objectbucketclaim_types.go

```diff
@@ -40,6 +39,9 @@ type ObjectBucketClaimSpec struct {
 	// AdditionalConfig gives providers a location to set
 	// proprietary config values (tenant, namespace, etc)
 	AdditionalConfig map[string]string `json:"additionalConfig"`
+	// ObjectBucketName is the name of the object bucket resource.  This is the authoritative
+	// determintaion for binding.
+	ObjectBucketName string
```

Comment:
Correct me if I'm wrong @tnielsen or @leseb, but the way that K8s and go structs with JSON work, if there is no json info given here, Kubernetes will never see the `ObjectBucketName` field, correct? It will treat it as though it doesn't exist I think.

Reply:
@copejon @jeffvance ^^

Reply:
Oh dang, yes this is bug.  To what degree it affects the controller I don't know.  I'll do some investigating.

Reply:
Really? I think it works just the json name is by default capitalized instead of `objectBucketName`...
Not sure if kubernetes is doing something different, but json will marshal it -
```
$ cat a.go
package main

import (
	"encoding/json"
	"fmt"
)

type A struct {
	Name             string `json:"xxxxxxxxxxx"`
	ObjectBucketName string
}

func main() {
	a := A{Name: "Jon", ObjectBucketName: "Cope"}
	s, _ := json.Marshal(a)
	fmt.Println(string(s))
}

$ go run a.go
{"xxxxxxxxxxx":"Jon","ObjectBucketName":"Cope"}
```

Reply:
thanks @guymguym.  So that's why it didn't produce to breaking behavior.  Wouldn't hurt to tag it for the sake of convention, but it's otherwise innocuous.